### PR TITLE
use useDailyEvent hook - Daily react hooks

### PR DESCRIPTION
**Pull request by harshithpabbati**
This PR is to use the useDailyEvent hook.

### CODE REVIEW

File: custom/shared/contexts/CallProvider.js [hunk 1]
```diff
@@ -124,14 +124,11 @@ export const CallProvider = ({
     setPreJoinNonAuthorized(requiresPermission && !token);
   }, [state, daily, token]);
 
-  useEffect(() => {
-    if (!daily) return;
-
-    if (cleanURLOnJoin)
-      daily.on('joined-meeting', () => router.replace(`/${room}`));
+  const handleOnJoinCleanUp = useCallback(() => {
+    if (cleanURLOnJoin) router.replace(`/${room}`);
```
Comment (jessmitch42) [inline custom/shared/contexts/CallProvider.js:128]: can you move this onto its own line (we typically only use single line if statements for returns). 
```
if (cleanURLOnJoin) {
    router.replace(`/${room}`);
}
```

**Review by jessmitch42 — APPROVED** [review]
LGTM! One request: can you have `fetch token` and `join as owner` set to `true` as the default for transcription? There's no user feedback in the UI that you need an owner token to use transcription so it's not super clear if it's broken or what. I think that little change will make it a lot more first-time-user friendly

**Comment by jessmitch42** [comment]
@harshithpabbati I noticed the join card doesn't have the toggles anymore. My feedback above was just to have the toggles set to true as the default value, not to remove them. I think we should still include the toggle inputs because it shows people that those values are meaningful via the UI. (Please feel free to ping me if any feedback isn't clear! 🙂 )
Screen Shot 2022-04-11 at 4 26 12 PM

**Comment by harshithpabbati** [comment]
> @harshithpabbati I noticed the join card doesn't have the toggles anymore. My feedback above was just to have the toggles set to true as the default value, not to remove them. I think we should still include the toggle inputs because it shows people that those values are meaningful via the UI. (Please feel free to ping me if any feedback isn't clear! 🙂 ) Screen Shot 2022-04-11 at 4 26 12 PM

Actually, we are doing it in this way for live-streaming and recording, so I thought it would be consistent to do it in the same way, do you think I should change it?

**Comment by jessmitch42** [comment]
@harshithpabbati my thinking is that a demo should be more explicit than a regular production app with how features work. It can be helpful to show that an owner token is needed for transcription (or live streaming or whatever) in the UI, so the user doesn't have to read through all the code to figure it out. We can add this when `examples` eventually gets a design update, though, since there are some more UX improvements we can do here.

**Review by jessmitch42 — APPROVED** [review]
nice work 👍

**Pull request merged**